micrometer-core/src/main/java/io/micrometer/core/instrument/internal/DefaultLongTaskTimer.java

```diff
@@ -91,11 +98,27 @@ public DefaultLongTaskTimer(Id id, Clock clock, TimeUnit baseTimeUnit,
 
     @Override
     public Sample start() {
-        SampleImpl sample = new SampleImpl();
-        activeTasks.add(sample);
+        final long startTime = clock.monotonicTime();
+        SampleImpl sample = new SampleImpl(startTime);
+        if (!activeTasks.add(sample)) {
+            sample = new SampleImplCounted(startTime, nextNonZeroCounter());
+            activeTasks.add(sample);
+        }
         return sample;
     }
 
+    private int nextNonZeroCounter() {
+        int nextCount;
+        while ((nextCount = counter.incrementAndGet()) == 0) {
```

Comment:
I'm not sure I understand why this (whole method) is needed. Could you explain?

Reply:
We should try to avoid `0` here because the classes are constructed such that `SampleImplCounted(startTime, 0)` compares equal to `SampleImpl(startTime)`, so the two would collide again.

This will happen on the 2^32-th collision on the same instance of `DefaultLongTaskTimer`, when `counter` has wrapped around the whole int range.

If you feel skipping 1 in 2^32 collisions is acceptable compared to having the strange code, I can just remove this method and the relative test